crates/turborepo-lib/src/engine/mod.rs

```diff
@@ -116,6 +119,12 @@ impl Engine<Built> {
         )
     }
 
+    // TODO get rid of static lifetime and figure out right way to tell compiler the
```

Comment:
I think what we'll want to eventually do is make an owned/borrowed version of task id ala https://github.com/sunshowers-code/borrow-complex-key-example.